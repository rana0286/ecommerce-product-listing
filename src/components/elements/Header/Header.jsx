import React from 'react';
import PropTypes from 'prop-types';

import './Header.scss';

const ProductListHeader = ({ children }) => {
  return (<div className="product-list-header">
    {children}
  </div>);
};

ProductListHeader.propTypes = {
  children: PropTypes.node,
};

ProductListHeader.defaultValue = {
  children: <span></span>,
};

export default ProductListHeader;
