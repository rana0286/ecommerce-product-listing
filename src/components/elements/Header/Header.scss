$header-background-color: #def1f4;

.product-list-header {
  align-items: center;
  background-color: $header-background-color;
  display: flex;
  justify-content: space-between;
  margin: 30px 0 10px;
  padding: 15px 10px;
}
