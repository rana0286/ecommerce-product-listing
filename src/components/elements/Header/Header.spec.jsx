import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';
import renderer from 'react-test-renderer';

describe('src/components/elements/Header', () => {
  it('Header component should render', () => {
    const component = shallow(<Header><a>anchor</a></Header>);
    const tree = renderer.
      create(<Header><h1>heading</h1></Header>)
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(component).toBeTruthy();
  });
});
