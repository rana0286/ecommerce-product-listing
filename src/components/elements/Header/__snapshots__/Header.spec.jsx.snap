// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/Header Header component should render 1`] = `
<div
  className="product-list-header"
>
  <h1>
    heading
  </h1>
</div>
`;
